vpr/src/draw/draw_basic.cpp

```diff
@@ -1103,6 +1103,81 @@ void draw_crit_path(ezgl::renderer* g) {
     }
 }
 
+/**
+ * @brief Draw critical path elements.
+ * 
+ * This function draws critical path elements based on the provided timing paths
+ * and indexes map. It is primarily used in server mode, where items are drawn upon request.
+ */
+void draw_crit_path_elements(const std::vector<tatum::TimingPath>& paths, const std::map<std::size_t, std::set<std::size_t>>& indexes, bool draw_crit_path_contour, ezgl::renderer* g) {
+    t_draw_state* draw_state = get_draw_state_vars();
+    const ezgl::color contour_color{0, 0, 0, 40};
+
+    auto draw_flyline_timing_edge_helper_fn = [](ezgl::renderer* renderer, const ezgl::color& color, ezgl::line_dash line_style, int line_width, float delay, 
+                                            const tatum::NodeId& prev_node, const tatum::NodeId& node, bool skip_draw_delays=false) {
+        renderer->set_color(color);
+        renderer->set_line_dash(line_style);
+        renderer->set_line_width(line_width);
+        draw_flyline_timing_edge(tnode_draw_coord(prev_node),
+                                tnode_draw_coord(node), delay, renderer, skip_draw_delays);
+
+        renderer->set_line_dash(ezgl::line_dash::none);
+        renderer->set_line_width(0);                        
+    };
+
+    for (const auto& [path_index, element_indexes]: indexes) {
+        if (path_index < paths.size()) {
+            const tatum::TimingPath& path = paths[path_index];
+
+            //Walk through the timing path drawing each edge
+            tatum::NodeId prev_node;
+            float prev_arr_time = std::numeric_limits<float>::quiet_NaN();
+            int element_counter = 0;
+            for (const tatum::TimingPathElem& elem : path.data_arrival_path().elements()) {
+                bool draw_current_element = element_indexes.empty() ? true : element_indexes.find(element_counter) != element_indexes.end();
```

Comment:
It is mostly personal preference, but I think this line is a little hard to understand. The following expression is a bit easier to read:
```C++
bool draw_current_element = element_indexes.empty() || element_indexes.find(element_counter) != element_indexes.end()
```

Reply:
fixed https://github.com/verilog-to-routing/vtr-verilog-to-routing/pull/2526/commits/32c0730dd2c0952ab15aeccdfa8070eb477fcd97